codalab/worker/docker_utils.py

```diff
@@ -268,3 +269,31 @@ def get_container_running_time(container):
     # formatted datetime string directly.
     container_running_time = parser.isoparse(end_time) - parser.isoparse(start_time)
     return container_running_time.total_seconds()
+
+
+@wrap_exception('Unable to check image size before pulling from Docker Hub')
+def check_image_size_before_pulling(image_spec):
```

Comment:
To make this more modular and reusable, I'd break this up into two functions, one that just gets the size of the image, and another one that checks the size.  In fact, the checking logic doesn't need to be in this file at all since it's not core to docker.

Reply:
Splitted the logic.